src/ForceIDPDiscoveryController.js

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) 2015-2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Spinner from 'views/shared/Spinner';
+import BaseLoginController from 'util/BaseLoginController';
+import Util from 'util/Util';
+
+export default BaseLoginController.extend({
+  className: 'force-idp-discovery',
+
+  View: Spinner,
+
+  Model: {},
+
+  initialize: function() {
+    let options = this.options;
+    const lastAuthResponse = options.appState.get('lastAuthResponse');
+    const stateToken = lastAuthResponse && lastAuthResponse?.stateToken;
+
+    const webfingerArgs = {
+      resource: 'okta:acct:undefined',
+      requestContext: stateToken,
```

Comment:
It cannot be undefined if we're ending up in this controller, it's guaranteed by the backend.

Reply:
yeah, this code originated from a IdpDiscovery model, in which case this might not be guaranteed, but in this controller it will be. I can remove the `?` part if you find that less confusing.

Reply:
should be okay.